packages-exp/auth-exp/src/core/mfa/mfa_info.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as externs from '@firebase/auth-types-exp';
+import {
+  PhoneMfaEnrollment,
+  MfaEnrollment
+} from '../../api/account_management/mfa';
+import { AuthErrorCode } from '../errors';
+import { fail } from '../util/assert';
+
+export abstract class MultiFactorInfo implements externs.MultiFactorInfo {
+  readonly uid: string;
+  readonly displayName?: string | null;
+  readonly enrollmentTime: string;
+
+  protected constructor(
+    readonly factorId: externs.ProviderId,
+    response: MfaEnrollment
+  ) {
+    this.uid = response.mfaEnrollmentId;
+    this.enrollmentTime = new Date(response.enrolledAt).toUTCString();
+    this.displayName = response.displayName;
+  }
+
+  static _fromServerResponse(
+    auth: externs.Auth,
```

Comment:
why is this accepting `externs.Auth` if it's an internal-only method? I think it would be best if all translation from external => internal types happens right at the edge, as soon as the object comes in

Reply:
done